bottypes/challenge.py

```diff
@@ -14,14 +16,17 @@ def __init__(self, ctf_channel_id, channel_id, name, category):
         self.players = {}
         self.is_solved = False
         self.solver = None
+        self.solve_date = 0
 
-    def mark_as_solved(self, solver_list):
+    def mark_as_solved(self, solver_list, solve_date=0):
         """
         Mark a challenge as solved.
         solver_list : List of usernames, that solved the challenge.
+        solve_date : Time of solve (epoch) (0: current time / value: set to specified value).
         """
         self.is_solved = True
         self.solver = solver_list
+        self.solve_date = solve_date if solve_date > 0 else int(time.time())
```

Comment:
@Grazfather Would it be a good idea to change `solve_date=0` to `solve_date = None` and do:
`self.solve_date = solve_date or int(time.time())`

Reply:
Yes.

Technically that would still work even if `solve_date=0`, but using `None` makes the intention more clear.

```python
>>> solve_date = 0
>>> import time
>>> x = solve_date or int(time.time())
>>> x
1513792933
```